### PR TITLE
Implement admin authenticator

### PR DESCRIPTION
**Pull request by 1995parham**
Admin authenticator will provides superuser access for its company users. It uses the same tenant concept in Soteria as other authenticators.

### CODE REVIEW

**Comment by 1995parham** [comment]
@mohammadiahmad @mehditeymorian code is not finished yet but if you can tell about your opinions on this, it would be great.

File: internal/authenticator/admin_authenticator.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		return fmt.Errorf("token is invalid: %w", err)
+	}
+
+	return nil
+}
+
+// ACL check a system user access to a topic.
+// because we returns is-admin: true, this endpoint shouldn't
+// be called.
+func (a AdminAuthenticator) ACL(
```
Comment (mohammadiahmad) [inline internal/authenticator/admin_authenticator.go:46]: @1995parham In long term view we can think about restricting services on their topics.

Comment (1995parham) [reply internal/authenticator/admin_authenticator.go:46]: You are right Ahmad jan, EMQ has a concept which creates tenants for you by prefixing topics. We can use this feature for supporting different vendors but right now we need shared topics. For example Snapp Box want to read the Snapp events this means we need to validate Snapp Box tokens for Snapp topics.

**Review by mehditeymorian — APPROVED** [review]
Everything seems well put together.🔥

**Pull request merged**